src/connection.rs

```diff
@@ -263,6 +318,67 @@ impl ActualConnection {
                     open: true,
                 })
             }
+            #[cfg(feature = "tls")]
+            ConnectionAddr::TcpTls {
+                ref host,
+                port,
+                insecure,
+            } => {
+                let tls_connector = if insecure {
+                    TlsConnector::builder()
+                        .danger_accept_invalid_certs(true)
+                        .danger_accept_invalid_hostnames(true)
+                        .use_sni(false)
+                        .build()?
+                } else {
+                    TlsConnector::new()?
+                };
+                let host: &str = &*host;
+                let tls = match timeout {
+                    None => {
+                        let tcp = TcpStream::connect((host, port))?;
+                        tls_connector.connect(host, tcp).unwrap()
+                    }
+                    Some(timeout) => {
+                        let mut tcp = None;
+                        let mut last_error = None;
+                        for addr in (host, port).to_socket_addrs()? {
+                            match TcpStream::connect_timeout(&addr, timeout) {
+                                Ok(l) => {
+                                    tcp = Some(l);
+                                    break;
+                                }
+                                Err(e) => {
+                                    last_error = Some(e);
+                                }
+                            };
+                        }
+                        match (tcp, last_error) {
+                            (Some(tcp), _) => tls_connector.connect(host, tcp).unwrap(),
+                            (None, Some(e)) => {
+                                fail!(e);
+                            }
+                            (None, None) => {
+                                fail!((
+                                    ErrorKind::InvalidClientConfig,
+                                    "could not resolve to any addresses"
+                                ));
+                            }
+                        }
+                    }
+                };
+                ActualConnection::TcpTls(TcpTlsConnection {
+                    reader: tls,
+                    open: true,
+                })
+            }
+            #[cfg(not(feature = "tls"))]
+            ConnectionAddr::TcpTls { .. } => {
+                fail!((
+                    ErrorKind::InvalidClientConfig,
+                    "Cannot connect to TCP with TLS without the tls feature"
```

Comment:
Oh I see. We already do this for unix socket as well, so that explains why it is happening further above as well. I'm ok with that then.